src/algebra/lie/basic.lean

```diff
@@ -544,90 +540,145 @@ end lie_algebra
 
 section lie_module
 
-variables (R : Type u) (L : Type v) [comm_ring R] [lie_ring L] [lie_algebra R L]
-variables (M  : Type v) [add_comm_group M] [module R M]
+variables (R : Type u) (L : Type v) (M  : Type w)
+variables [comm_ring R] [lie_ring L] [lie_algebra R L] [add_comm_group M] [module R M]
+
+class lie_ring_module extends has_bracket L M :=
+(add_lie : ∀ (x y : L) (m : M), ⁅x + y, m⁆ = ⁅x, m⁆ + ⁅y, m⁆)
+(lie_add : ∀ (x : L) (m n : M), ⁅x, m + n⁆ = ⁅x, m⁆ + ⁅x, n⁆)
+(lie_lie : ∀ (x y : L) (m : M), ⁅⁅x, y⁆, m⁆ = ⁅x, ⁅y, m⁆⁆ - ⁅y, ⁅x, m⁆⁆)
 
 /-- A Lie module is a module over a commutative ring, together with a linear action of a Lie
 algebra on this module, such that the Lie bracket acts as the commutator of endomorphisms. -/
-class lie_module extends linear_action R L M :=
-(lie_act : ∀ (l l' : L) (m : M), act ⁅l, l'⁆ m = act l (act l' m) - act l' (act l m))
-
-@[simp] lemma lie_act [lie_module R L M]
-  (l l' : L) (m : M) : linear_action.act R ⁅l, l'⁆ m =
-                       linear_action.act R l (linear_action.act R l' m) -
-                       linear_action.act R l' (linear_action.act R l m) :=
-  lie_module.lie_act l l' m
-
-protected lemma of_endo_map_action (α : L →ₗ⁅R⁆ module.End R M) (x : L) (m : M) :
-  @linear_action.act R _ _ _ _ _ _ _ (linear_action.of_endo_map R L M α) x m = α x m := rfl
+class lie_module [lie_ring_module L M] :=
```

Comment:
Why do you assume `lie_ring_module` as argument, instead of extending it?

Reply:
Because `lie_module` has more carrier types (`R`, `L`, `M`) than `lie_ring_module` (`L`, `M`) so we would have the recursive typeclass search problem if I extended: I think we'd have loops where the algorithm spawns metavariables for `R` recursively when trying to find a `lie_ring_module` instance.

I.e., the same reason e.g., a regular `module` cannot extend `add_comm_group` but has to have it as an unbundled argument.